### DISCUSSION TITLE
STC-ISP Initializing

### DISCUSSION THREAD

**Discussion by NeoCHoZen** [danielnilsson9/bbs-fw#36] [General]
Aloha Daniel, 

I'm in the process of flashing your BBSHD firmware. I have the steps down pretty good and I feel confident I won't screw anything up, but one question just to make sure....

- when launching the STC-ISP application, two windows open up. 1 is the STC UI and the other window is the "Please wait while initializing".
- That please wait window never does away. 

So do I need to keep waiting? It's been 15+ minutes as of the writing of this post, or can I just start the process per your instructions?

Any info would be appreciated as I only have this one bike and I can't afford to brick anything. 

Mahalo,
Tim

**Comment by danielnilsson9** [comment]
Hi

I'm not sure I have ever seen the "initializing" window. Which version of STC-ISP do you got?

**Reply by NeoCHoZen** [reply]
V6.90D. Screenshot attached. The initializing window never goes away. I can only assume in general that there's some environment condition not being met.


![2022-08-09_10h11_14](https://user-images.githubusercontent.com/110883915/183751682-d341af83-09e7-4132-8aa5-3fc2029f7acc.png)

**Reply by danielnilsson9** [reply]
I see, that seems to be a new window in newer versions of this software. Download 6.88 from the linked website in the flashing guide and do not update and you should be fine. I can confirm I get the same behavior when running 6.90E.

**Reply by NeoCHoZen** [reply]
Thank you Daniel! I did the same assumption and reverted back to 6.88 and I'm good to execute. I'm nervous about doing this, and I hope I don't f stuff up. 🤙

**Reply by danielnilsson9** [reply]
You are not likely to break anything, it's pretty hard as the original firmware can always be flashed again. There is only one dangerous option, the one that is called "Next time program only when P3.2 and P3.3 is...". DO NOT check that box or you might have to open up and depot your controller in order to be able to flash it again.

**Reply by NeoCHoZen** [reply]
Duly noted!

**Reply by dav0000000** [reply]
When that 'Please wait while inititaling" window displays in the new STC micro just hit ESC and it will disappear.
